api/server/router/build/build_routes.go

```diff
@@ -71,11 +71,13 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 	options.RemoteContext = r.FormValue("remote")
 	if versions.GreaterThanOrEqualTo(version, "1.32") {
 		apiPlatform := r.FormValue("platform")
-		p := system.ParsePlatform(apiPlatform)
-		if err := system.ValidatePlatform(p); err != nil {
-			return nil, errdefs.InvalidParameter(errors.Errorf("invalid platform: %s", err))
+		if len(strings.TrimSpace(apiPlatform)) != 0 {
+			m, err := platforms.Parse(apiPlatform)
+			if err != nil {
+				return nil, err
```

Comment:
~~Should this return an `errdefs.InvalidParameter(()`?~~

Nevermind, looks like `platforms.Parse()` already implements that interface

Reply:
Right